lib/segment/src/vector_storage/in_ram_persisted_vectors.rs

```diff
@@ -0,0 +1,117 @@
+use std::path::{Path, PathBuf};
+
+use common::types::PointOffsetType;
+use num_traits::AsPrimitive;
+
+use crate::common::operation_error::OperationResult;
+use crate::common::Flusher;
+use crate::vector_storage::chunked_mmap_vectors::ChunkedMmapVectors;
+use crate::vector_storage::chunked_vector_storage::ChunkedVectorStorage;
+use crate::vector_storage::chunked_vectors::ChunkedVectors;
+
+#[derive(Debug)]
+pub struct InRamPersistedVectors<T: Sized + 'static> {
+    mmap_storage: ChunkedMmapVectors<T>,
+    vectors: ChunkedVectors<T>,
+}
+
+impl<T: Sized + Copy + Clone + Default + 'static> InRamPersistedVectors<T> {
+    pub fn open(directory: &Path, dim: usize) -> OperationResult<Self> {
+        let mmap_storage = ChunkedMmapVectors::open(directory, dim, Some(false))?;
+
+        let mut vectors = ChunkedVectors::new(dim);
+
+        let total_vectors = mmap_storage.len();
+
+        for key in 0..total_vectors {
```

Comment:
because chunk size is equal, can we directly copy the whole file to the chunk without `push`? It might be faster on load

Reply:
Theoretically yes, but it will break abstraction